src/Plot.elm

```diff
@@ -0,0 +1,107 @@
+module Plot where
+
+import Svg exposing (svg, Svg)
+import Private.Extras.SvgAttributes exposing (width, height)
+import Private.Dimensions exposing (Dimensions)
+import Private.Margins as Margins exposing (Margins)
+import Private.BoundingBox as BoundingBox exposing (BoundingBox)
+import Private.Scale.Utils as Scale
+import Private.Scale exposing (Scale)
+import Plot.Axis as Axis exposing (Axis)
+import Private.Bars as Bars exposing (Orient)
+import Private.Axis.View as AxisView
+import Private.Title as Title
+
+type alias Plot =
+  { dimensions: Dimensions
+  , margins: Margins
+  , svgs: List (BoundingBox -> List Svg)
+  , eventHandlers: List (BoundingBox -> Svg.Attribute)
+  , attrs: List Svg.Attribute
+  , title : Title.Model
+  }
+
+createPlot : Float -> Float -> Plot
+createPlot w h =
+  { dimensions = { width = w, height = h }
+  , margins = Margins.init
+  , svgs = []
+  , eventHandlers = []
+  , attrs = [width w, height h]
+  , title = Title.init
+  }
+
+addTitle : String -> List Svg.Attribute -> Plot -> Plot
+addTitle title attrs plot =
+  { plot | title = Title.create title attrs }
+
+attributes : List Svg.Attribute -> Plot -> Plot
+attributes attrs plot =
+  { plot | attrs = plot.attrs ++ attrs }
```

Comment:
I don't understand what you are mean by "you'll always have to reconstruct the model for a Plot every time you need to change or remove attributes". This function allows you to add top level event handlers, styling, etc. There is no removing of attributes. Both of the examples provided are also equivalent to what is already there.